### PR TITLE
PSR-1232 Added members T1 page

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: test/controllers/nonsipp/memberdetails/SchemeMembersListControllerSpec.scala [hunk 1]
```diff
@@ -16,25 +16,51 @@
 
 package controllers.nonsipp.memberdetails
 
-import pages.nonsipp.memberdetails._
+import services.PsrSubmissionService
```
Comment (kberec) [inline test/controllers/nonsipp/memberdetails/SchemeMembersListControllerSpec.scala:19]: Looks PR builder failed when compiling tests.

File: app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala [hunk 2]
```diff
@@ -61,34 +67,88 @@ class SchemeMembersListController @Inject()(
...
+                mode,
+                filteredMembers,
+                viewOnlyUpdated = (mode, request.previousUserAnswers) match {
+                  case (ViewOnlyMode, Some(previousUserAnswers)) =>
+                    val updated = getCompletedOrUpdatedTaskListStatus(
+                      request.userAnswers,
+                      previousUserAnswers,
+                      pages.nonsipp.memberdetails.Paths.personalDetails
+                    ) == Updated
+                    logger.info(s"""[ViewOnlyMode] Status for member details list is ${if (updated) "updated"
```
Comment (kberec) [inline app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala:131]: Do you want to keep these logs?

File: app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala [hunk 3]
```diff
@@ -225,7 +337,45 @@ object SchemeMembersListController {
       refresh = None,
       Message("site.saveAndContinue"),
       None,
-      onSubmit = routes.SchemeMembersListController.onSubmit(srn, page, manualOrUpload)
+      onSubmit = routes.SchemeMembersListController.onSubmit(srn, page, manualOrUpload),
+      mode = mode,
+      optViewOnlyDetails = Option.when(mode == ViewOnlyMode) {
```
Comment (kberec) [inline app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala:342]: Could be `isViewOnlyMode` here too, as you added it in this PR

File: app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala [hunk 4]
```diff
@@ -100,14 +100,11 @@ class PclsMemberListController @Inject()(
                 updatedUserAnswers <- buildUserAnswerBySelection(srn, value, optionList.flatten.size)
                 _ <- saveService.save(updatedUserAnswers)
                 submissionResult <- if (value) {
-                  psrSubmissionService.submitPsrDetails(
+                  psrSubmissionService.submitPsrDetailsWithUA(
```
Comment (kberec) [inline app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala:103]: There are other places too.... Do you think it would be safer to deprecate `submitPsrDetails` in favour of `submitPsrDetailsWithUA`? Not as a part of this ticket, but separately...

File: app/transformations/MemberPaymentsTransformer.scala [hunk 5]
```diff
@@ -135,12 +135,29 @@ class MemberPaymentsTransformer @Inject()(
...
       val memberDetailsWithCorrectVersion: List[MemberDetails] = memberDetailsWithCorrectState.map {
         case (index, currentMemberDetail) =>
           val optInitialMemberDetail = initialMemberDetails.get(index)
+          val refinedCurrentMember =
+            currentMemberDetail
+              .copy(
+                memberLumpSumReceived =
+                  if (currentMemberDetail.memberLumpSumReceived.exists(_.empty)) None
+                  else currentMemberDetail.memberLumpSumReceived
+//                pensionAmountReceived =
```
Comment (kberec) [inline app/transformations/MemberPaymentsTransformer.scala:148]: left comments and printlns

File: app/transformations/MemberPaymentsTransformer.scala [hunk 6]
```diff
@@ -135,12 +135,29 @@ class MemberPaymentsTransformer @Inject()(
       }
 
       // Omit memberPSRVersion if member has changed
+      // Check for empty member payment sections before comparing members
       val memberDetailsWithCorrectVersion: List[MemberDetails] = memberDetailsWithCorrectState.map {
         case (index, currentMemberDetail) =>
           val optInitialMemberDetail = initialMemberDetails.get(index)
+          val refinedCurrentMember =
```
Comment (kberec) [inline app/transformations/MemberPaymentsTransformer.scala:142]: This change (and related changes in other files) seem not related to the ticket, is it intentional to have it in this PR?

File: app/views/ListView.scala.html [hunk 7]
```diff
@@ -15,79 +15,107 @@
...
+        govukButton: GovukButton,
+        govukRadios: GovukRadios,
+        hmrcListWithActions: HmrcListWithActions,
+        govukInsetText: GovukInsetText,
+        govukPagination: GovukPagination
 )
 @(form: Form[Boolean], viewmodel: FormPageViewModel[ListViewModel])(implicit request: Request[_], messages: Messages)
 
-@layout(pageTitle = title(form, viewmodel.title.toMessage)) {
+    @layout(pageTitle = title(form, viewmodel.title.toMessage)) {
```
Comment (kberec) [inline app/views/ListView.scala.html:29]: Do we need a selection here for the view-only title? (as it is currently duplicated and if someone in future passes in separate values from the controller)

File: test/controllers/nonsipp/memberdetails/SchemeMembersListControllerSpec.scala [hunk 8]
```diff
@@ -172,5 +206,89 @@ class SchemeMembersListControllerSpec extends ControllerBaseSpec {
...
+                optYear = Some(yearString),
+                optCurrentVersion = Some(submissionNumberTwo),
+                optPreviousVersion = Some(submissionNumberOne),
+                compilationOrSubmissionDate = Some(submissionDateTwo)
+              )
+            )
+        }.withName("OnPageLoadViewOnly renders ok with no changed flag")
+      )
+
+      val updatedUserAnswers = currentUserAnswers
```
Comment (kberec) [inline test/controllers/nonsipp/memberdetails/SchemeMembersListControllerSpec.scala:247]: I assume this is still to do, i.e. to test viewOnlyUpdated = true...

File: app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala (hunk 2 above)
Comment (westwater) [reply app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala:131]: Yeah just incase we have issues in the future QA's can supply FE logs to narrow down the issue

File: app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala (hunk 4 above)
Comment (westwater) [reply app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala:103]: Yeah we need to, it's caused too many bugs already

File: app/transformations/MemberPaymentsTransformer.scala (hunk 6 above)
Comment (westwater) [reply app/transformations/MemberPaymentsTransformer.scala]: Nope accidentally added things from another branch here, removed now

File: app/views/ListView.scala.html (hunk 7 above)
Comment (westwater) [reply app/views/ListView.scala.html]: added

File: app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala (hunk 4 above)
Comment (kberec) [reply app/controllers/nonsipp/memberreceivedpcls/PclsMemberListController.scala:145]: Just to check if my understanding is correct before I create a ticket...
In this case it was actually working correctly, because of the implicit parameter few lines down in the previous version of the code: `request = DataRequest(request.request, updatedUserAnswers)`.
It only causes bugs if `request` is not updated like that.
Is this understanding correct?

**Review by kberec — APPROVED** [review]
Looks good, just check please if the PR builder passes before merging it.
